pkg/controller/backup_controller.go

```diff
@@ -299,9 +336,121 @@ func (c *backupController) processBackup(key string) error {
 		log.WithError(err).Error("error updating backup's final status")
 	}
 
+	if request.Status.Phase == velerov1api.BackupPhaseUploading || request.Status.Phase == velerov1api.BackupPhaseUploadingPartialFailure {
+		removeFromTracker = false // Leave it in the tracker so it gets checked
+		// Set NextCheckTime here so that checkAndFinalizeBackupsLoop will pick it up now, after everything has been
+		// persisted otherwise we have a race
+		request.NextCheckTime = time.Now().Add(c.uploadProgressCheckInterval)
+	}
+
 	return nil
 }
 
+// Execute periodically to check upload progress for backups that are in Uploading or UploadingPartialFailure phase
+func (c *backupController) checkAndFinalizeBackupsLoop() {
+	for true {
+		time.Sleep(time.Second) // Wake up frequently so that we will process any backups moving into Upload phase quickly
+		// This wakeup is pretty lightweight as it just looks for things to process, we actually
+		// check progress less frequently, by default once per minute per backup
+		uploadingBackups := c.backupTracker.GetByPhase(velerov1api.BackupPhaseUploading)
+		uploadingBackups = append(uploadingBackups, c.backupTracker.GetByPhase(velerov1api.BackupPhaseUploadingPartialFailure)...)
+		for _, curBackup := range uploadingBackups {
+			if curBackup.NextCheckTime != dontProcessTime && curBackup.NextCheckTime.After(time.Now()) {
```

Comment:
Well, it makes it explicit that 0 is dontProcess and we shouldn't change it so that it does start processing.  I think it makes the code clearer, you don't have to guess when things won't be processed.